src/classes/frUtil.cls

```diff
@@ -95,8 +95,22 @@ public class frUtil {
         logError(errObject, recordId, error);
     }
     
+    private static Set<StatusCode> duplicateValueStatusCodes = 
+        new Set<StatusCode>{StatusCode.DUPLICATE_EXTERNAL_ID, StatusCode.DUPLICATE_VALUE};
+            
     public static void logException(Entity errObject, String recordId, Exception ex) {
-        logError(errObject, recordId, 'Operation failed. Exception: '+ex.getMessage());
+        if(ex instanceof DMLException) {
+            DMLException dmlEx = (DMLException)ex;
+            //If it's the race condition on fr_Id__c duplicate value, then ignore it.  Else, log it
+            for(Integer i = 0; i < dmlEx.getNumDml(); i++) {
+                if(!(dmlEx.getDmlMessage(i).contains('fr_Id__c') && duplicateValueStatusCodes.contains(dmlEx.getDmlType(i)))) {
```

Comment:
Is 'fr_Id__c' something we define in an enum or even static String somewhere? I know it's probably unlikely to change but still best if we have an object value that is shared rather than a string literal.

Reply:
Hmm interesting point.  This is a custom field that we have defined on every object that we integrate with Funraise and we named it the same thing in each instance.  So we could reference one of the fields here OR throw this string it in a static constant somewhere.  Do you have a preference?

Reply:
I also updated this `contains` -> `containsIgnoreCase`

Reply:
I think in that case a static constant makes more sense